### PR TITLE
Bug 1899350: Include dhcp client-id and bond options during ovs-configuration

### PR DESCRIPTION
**Pull request by trozet**
Adds copying over the dhcp client-id as well as bond options from the interface being migrated over to OVS.

### CODE REVIEW

**Comment by trozet** [comment]
@danwinship @dcbw

**Comment by cgwalters** [comment]
I think it'd be pretty easy BTW to:

- Move this logic into a container in the SDN
- Change the MCO templates to pull down and execute that container via podman before starting kubelet, much like we do with `machine-config-daemon-firstboot.service`

**Comment by danwinship** [comment]
(Filed a jira card about moving the logic out of MCO: https://issues.redhat.com/browse/SDN-1374)

File: templates/common/_base/files/configure-ovs-network.yaml
Comment (danwinship) [inline templates/common/_base/files/configure-ovs-network.yaml:138]: This is going to have quoting issues. The `bond.options` value can definitely have spaces in it, and maybe other shell metacharacters.

**Review by danwinship — CHANGES_REQUESTED** [review]

File: templates/common/_base/files/configure-ovs-network.yaml
Comment (celebdor) [inline templates/common/_base/files/configure-ovs-network.yaml:88]: No need to both grep and awk
```suggestion
      old_conn=$(nmcli --fields UUID,DEVICE conn show --active | awk "/${iface}/ {print \$1}")
```

File: templates/common/_base/files/configure-ovs-network.yaml
Comment (celebdor) [inline templates/common/_base/files/configure-ovs-network.yaml:96]: Do we also need to preserve `ipv6.dhcp-duid` ?

File: templates/common/_base/files/configure-ovs-network.yaml
Comment (trozet) [reply templates/common/_base/files/configure-ovs-network.yaml:138]: I don't think bond.options can have spaces:
```
[trozet@trozet ~]$ nmcli conn add type bond con-name bond1 bond.options "mode=active-backup miimon=100 fail_over_mac=follow"
Error: failed to modify bond.options: 'active-backup miimon=100 fail_over_mac=follow' not among [balance-rr, active-backup, balance-xor, broadcast, 802.3ad, balance-tlb, balance-alb].
[trozet@trozet ~]$ 
[trozet@trozet ~]$ nmcli conn add type bond con-name bond1 bond.options mode=active-backup miimon=100 fail_over_mac=follow
Error: invalid <setting>.<property> 'miimon=100'.
[trozet@trozet ~]$ nmcli conn add type bond con-name bond3 bond.options mode=active-backup,miimon=100,fail_over_mac=follow
Connection 'bond3' (4fefd985-fbb9-4d16-a51d-fd28361e5cae) successfully added.
```

ifcfg bond options can have spaces:
BONDING_OPTS="mode=active-backup miimon=100 fail_over_mac=follow"

however when the ifcfg is loaded into NM, NM is smart enough to convert it to csv

Comment (danwinship) [reply templates/common/_base/files/configure-ovs-network.yaml:138]: after some experimentation it looks like this should be ok

File: templates/common/_base/files/configure-ovs-network.yaml
Comment (trozet) [reply templates/common/_base/files/configure-ovs-network.yaml:96]: yeah will add

File: templates/common/_base/files/configure-ovs-network.yaml
Comment (trozet) [reply templates/common/_base/files/configure-ovs-network.yaml:88]: this made me realize we probably need to be more deterministic here how we match the interface. If iface was eth1, and we had eth10 on the system, we would accidentally grab the wrong one. Going to change it to something like:
```
nmcli --fields UUID,DEVICE conn show --active | awk "/\s${iface}\s*\$/ {print \$1}"
```

**Comment by kikisdeliveryservice** [comment]
is this ovn test supposed to pass @trozet 

/hold

**Review by kikisdeliveryservice — APPROVED** [review]
approving, tim if you don't need that test to pass remove the hold

**Comment by trozet** [comment]
the test got far enough that it made it past what this patch invokes, so the failure isn't due to this

/hold cancel

**Pull request merged**
